python/ray/data/dataset.py

```diff
@@ -2002,15 +2002,15 @@ def take_all(self, limit: int = 100000) -> List[T]:
         Returns:
             A list of all the records in the dataset.
         """
+        if limit is not None:
+            raise ValueError(
+                'take_all() does not accept "limit" argument any more. take_all() '
+                f"will take entire dataset. Use take(limit={limit}) instead when you "
+                "want to set a limit."
+            )
```

Comment:
Throw this error instead of removing `limit` argument completely for 2.2.
Just in case, we do have some users depending on `take_all(limit=N)`. This is to guide them migrate to use `take(limit=N)`.